valqueries-test/src/test/java/com/valqueries/automapper/AutoMapperBaseTests.java

```diff
@@ -1223,6 +1225,265 @@ public void update_withoutPreviousRecord() {
 		assertEquals(0, affectedRows);
 	}
 
+	@Test
+	@TestClasses(Car.class)
+	public void insert_happy() {
+		Car model = factory.get(Car.class);
+		model.setId(UUID.randomUUID());
+		model.setTitle("Muh");
+		model.setBrand(Brand.Porsche);
+		model.setCreatedAt(ZonedDateTime.now().withZoneSameInstant(ZoneOffset.UTC).truncatedTo(ChronoUnit.SECONDS));
+		carRepository.doRetryableInTransaction(tx -> carRepository.insert(tx, model));
+
+		Optional<Car> actualOptional = carRepository.get(model.getId());
+		Car actual = actualOptional.orElseThrow(RuntimeException::new);
```

Comment:
this is a minor thing but why you throwing RuntimeException? why is this necessary? by default if you try to get it will throw an `NoSuchElementException`. If its to better assert this, why not something like:

```suggestion
		if(!actualOptional.isPresent()){
			fail("Car id not found in database");
		}
		Car actual = actualOptional.get();
```